internal/server/server.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+// Version information, assigned by ldflags
+var (
+	CommitHash string
+	BuildDate  string
+)
+
+// Server is the wrapper of http.Server
+type Server struct {
+	server *http.Server
+}
+
+// New creates a new Bifrost server
+func New(port int) *Server {
+	r := mux.NewRouter()
```

Comment:
Even mux won't be required I think. It's just going to be plain request interception from `/`.